src/ConnectedMode/Suppressions/TimedUpdateHandler.cs

```diff
@@ -39,33 +40,56 @@ internal sealed class TimedUpdateHandler : IDisposable
         private readonly IServerHotspotStoreUpdater serverHotspotStoreUpdater;
         private readonly IQualityProfileUpdater qualityProfileUpdater;
         private readonly ILogger logger;
+        private readonly IActiveSolutionBoundTracker activeSolutionBoundTracker;
 
         private bool disposed;
 
         [ImportingConstructor]
         public TimedUpdateHandler(ISuppressionIssueStoreUpdater suppressionIssueStoreUpdater,
             IServerHotspotStoreUpdater serverHotspotStoreUpdater,
             IQualityProfileUpdater qualityProfileUpdater,
-            ILogger logger)
-            : this(suppressionIssueStoreUpdater, serverHotspotStoreUpdater, qualityProfileUpdater, logger, new TimerFactory()) { }
+            ILogger logger,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker)
+            : this(suppressionIssueStoreUpdater, serverHotspotStoreUpdater, qualityProfileUpdater, logger, activeSolutionBoundTracker, new TimerFactory()) { }
 
         internal /* for testing */ TimedUpdateHandler(ISuppressionIssueStoreUpdater suppressionIssueStoreUpdater,
             IServerHotspotStoreUpdater serverHotspotStoreUpdater,
             IQualityProfileUpdater qualityProfileUpdater,
             ILogger logger,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
             ITimerFactory timerFactory)
         {
             this.suppressionIssueStoreUpdater = suppressionIssueStoreUpdater;
             this.serverHotspotStoreUpdater = serverHotspotStoreUpdater;
             this.qualityProfileUpdater = qualityProfileUpdater;
             this.logger = logger;
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
 
             refreshTimer = timerFactory.Create();
             refreshTimer.AutoReset = true;
             refreshTimer.Interval = MillisecondsToWaitBetweenRefresh;
             refreshTimer.Elapsed += OnRefreshTimerElapsed;
 
-            refreshTimer.Start();
+            activeSolutionBoundTracker.SolutionBindingChanged += SolutionBindingChanged;
+
+            CreateTimerIfInConnectedMode(activeSolutionBoundTracker.CurrentConfiguration);
+        }
+
+        private void SolutionBindingChanged(object sender, ActiveSolutionBindingEventArgs activeSolutionBindingEventArgs)
+        {
+            CreateTimerIfInConnectedMode(activeSolutionBindingEventArgs.Configuration);
+        }
+
+        private void CreateTimerIfInConnectedMode(BindingConfiguration bindingConfiguration)
+        {
+            if (bindingConfiguration?.Mode != SonarLintMode.Standalone)
```

Comment:
Instead of this we could also have a solution in which we create a new instance of the refreshTimer but I don't think it really makes sense considering we can just stop and start it

Reply:
Agreed; but the name of the method should change - `CreateTimer...` is a bit misleading.
It's not obvious what to call the method, though: `SetTimerStatus`, `EnableTimerInConnectedMode`, `StartStopTimer`?